mesh/src/vertex/MutableVertexMesh.cpp

```diff
@@ -1278,21 +1278,163 @@ void MutableVertexMesh<ELEMENT_DIM, SPACE_DIM>::IdentifySwapType(Node<SPACE_DIM>
+                        *     x     the adjoining edges (x's)
+                        *
+                        *
+                        */
+
+                        // First remove void
+                        // Find all three nodes in void
+                        Node<SPACE_DIM>* p_node_C = this->mNodes[next_node_2]; // The other node in the triangular void
+                        if (next_node_1 == previous_node_2)
+                        {
+                            p_node_C = this->mNodes[next_node_1];
+                        }
+
+                       /*
+                        * In two steps, merge nodes A, B and C into a single node.  This is implemented in such a way that
+                        * the ordering of their indices does not matter.
+                        */
+
+                        PerformNodeMerge(pNodeA, pNodeB);
+
+                        Node<SPACE_DIM>* p_merged_node = pNodeB;
+
+                        if (pNodeB->IsDeleted())
+                        {
+                            p_merged_node = pNodeA;
+                        }
+
+                        PerformNodeMerge(p_node_C, p_merged_node);
+
+                        if (p_merged_node->IsDeleted())
+                        {
+                            p_merged_node = p_node_C;
+                        }
+
+                        // Tag remaining node as non-boundary
+                        p_merged_node->SetAsBoundaryNode(false);
+
+                        // Now merge this node with one of the nearest vertices keeping that vertices location.
+                        Node<SPACE_DIM>* p_end_node; // The neighbouring vertex to merge to
+
+                        std::set<unsigned> previous_previous_elem_indices = this->mNodes[previous_previous_node_1]->rGetContainingElementIndices();
+
+                        std::set<unsigned> shared_elements;
+                        std::set_intersection(all_indices.begin(),
+                              all_indices.end(),
+                              previous_previous_elem_indices.begin(),
+                              previous_previous_elem_indices.end(),
+                              std::inserter(shared_elements, shared_elements.begin()));
+
+                        assert(shared_elements.size()<3);
+
+                        if (shared_elements.size()==2)
+                        {
+                            //This neighbouring node is in the same 2 elements so treat this as end node
+                            p_end_node = this->mNodes[previous_previous_node_1];
+                        }
+                        else
+                        {
+                            /*
+                             * If this trips then the adjacent node isn't in both elements.
+                             * So check the other adjacent node by uncommenting the code blow and adding a new test
```

Comment:
@jmosborne there is no code to uncomment below this line - could you check whether this comment makes sense and remove/alter it if not?